worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -200,36 +205,25 @@ private void deleteExistingDirectory(Path outputDirPath) {
             if (recordsProcessedCount % cancellationCheckFrequency == 0) {
                 errorCount += processHandles(futureHandles);
 
-                // A Job could run for a long time, perhaps hours.
-                // While the job is in progress, the job could be cancelled.
-                // So the worker needs to periodically check the job status to ensure it has not been cancelled.
-
-                jobStatus = jobRepository.findJobStatus(jobUuid);
-                if (jobHasBeenCancelled(jobStatus)) {
+                isCancelled = hasJobBeenCancelled(jobUuid);
+                if (isCancelled) {
                     log.warn("Job [{}] has been cancelled. Attempting to stop processing the job shortly ... ", jobUuid);
+                    cancelFuturesInQueue(futureHandles);
                     break;
                 }
             }
         }
 
-        if (jobHasBeenCancelled(jobStatus)) {
-            final String errMsg = "Job was cancelled while it was being processed";
-            log.warn("{} - JobUuid :[{}]", errMsg, jobUuid);
-
-            // cancel any outstanding futures that have not started processing.
-            futureHandles.parallelStream().forEach(future -> future.cancel(false));
-
-            //At this point, there may be a few futures that are already in progress.
-            //But all the futures that are not yet in progress would be cancelled.
-
-            throw new JobCancelledException(errMsg);
-        }
-
         while (!futureHandles.isEmpty()) {
             sleep();
             errorCount += processHandles(futureHandles);
         }
 
+        if (isCancelled) {
+            final String errMsg = "Job was cancelled while it was being processed";
+            log.warn("{} - JobUuid :[{}]", errMsg, jobUuid);
```

Comment:
jobUUid should already be present from MDC.put in JobHandler

Reply:
done - removed.